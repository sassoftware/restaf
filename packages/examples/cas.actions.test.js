/*
 * ------------------------------------------------------------------------------------
 *   Copyright (c) SAS Institute Inc.
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 * ---------------------------------------------------------------------------------------
 *
 */

'use strict';
let testFunctions = require('./testFunctions');
let testInfo;
beforeAll(async () => {
	testInfo = await require('./lib/setupAll')();
});

test('CAS Session', async () => {

	let r = await testFunctions.casSession(testInfo);
	expect(r).toBe('done');
});

test('CAS Echo', async () => {

	let r = await testFunctions.casEcho(testInfo);
	expect(r).toBe('done');
});

test('CAS DataStep and Fetch', async () => {
	let r = await testFunctions.casDSandFetch(testInfo);
	expect(r).toMatchSnapshot();
});

test('CAS Tables', async () => {
	let r = await testFunctions.casTables(testInfo);
	expect(r).toBe('done');
});

test('CAS upload csv', async () => {
	let r = await testFunctions.casUploadCsv(testInfo);
	expect(r).toMatchSnapshot();
});
	