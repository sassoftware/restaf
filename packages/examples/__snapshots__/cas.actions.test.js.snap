// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAS DataStep and Fetch 1`] = `
Object {
  "_ctb": true,
  "attributes": Object {
    "Action": Object {
      "type": "string",
      "value": "fetch",
    },
    "Actionset": Object {
      "type": "string",
      "value": "table",
    },
    "CreateTime": Object {
      "type": "double",
      "value": 0,
    },
  },
  "label": "Selected Rows from Table SCORE",
  "name": "Fetch",
  "rows": Array [
    Array [
      1,
      10,
      20,
      30,
      60,
    ],
  ],
  "schema": Array [
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "_Index_",
      "type": "int",
      "width": 4,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "x1",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "x2",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "x3",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "score1",
      "type": "double",
      "width": 8,
    },
  ],
  "title": "Selected Rows from Table SCORE",
}
`;

exports[`CAS upload csv 1`] = `
Object {
  "_ctb": true,
  "attributes": Object {
    "Action": Object {
      "type": "string",
      "value": "fetch",
    },
    "Actionset": Object {
      "type": "string",
      "value": "table",
    },
    "CreateTime": Object {
      "type": "double",
      "value": 0,
    },
  },
  "label": "Selected Rows from Table CARSJEST",
  "name": "Fetch",
  "rows": Array [
    Array [
      1,
      "Acura",
      "MDX",
      "SUV",
      "Asia",
      "All",
      36945,
      33337,
      3.5,
      6,
      265,
      17,
      23,
      4451,
      106,
      189,
    ],
    Array [
      2,
      "Acura",
      "RSX Type S 2dr",
      "Sedan",
      "Asia",
      "Front",
      23820,
      21761,
      2,
      4,
      200,
      24,
      31,
      2778,
      101,
      172,
    ],
    Array [
      3,
      "Acura",
      "TSX 4dr",
      "Sedan",
      "Asia",
      "Front",
      26990,
      24647,
      2.4,
      4,
      200,
      22,
      29,
      3230,
      105,
      183,
    ],
    Array [
      4,
      "Acura",
      "TL 4dr",
      "Sedan",
      "Asia",
      "Front",
      33195,
      30299,
      3.2,
      6,
      270,
      20,
      28,
      3575,
      108,
      186,
    ],
    Array [
      5,
      "Acura",
      "3.5 RL 4dr",
      "Sedan",
      "Asia",
      "Front",
      43755,
      39014,
      3.5,
      6,
      225,
      18,
      24,
      3880,
      115,
      197,
    ],
    Array [
      6,
      "Acura",
      "3.5 RL w/Navigation 4dr",
      "Sedan",
      "Asia",
      "Front",
      46100,
      41100,
      3.5,
      6,
      225,
      18,
      24,
      3893,
      115,
      197,
    ],
    Array [
      7,
      "Acura",
      "NSX coupe 2dr manual S",
      "Sports",
      "Asia",
      "Rear",
      89765,
      79978,
      3.2,
      6,
      290,
      17,
      24,
      3153,
      100,
      174,
    ],
    Array [
      8,
      "Audi",
      "A4 1.8T 4dr",
      "Sedan",
      "Europe",
      "Front",
      25940,
      23508,
      1.8,
      4,
      170,
      22,
      31,
      3252,
      104,
      179,
    ],
    Array [
      9,
      "Audi",
      "A41.8T convertible 2dr",
      "Sedan",
      "Europe",
      "Front",
      35940,
      32506,
      1.8,
      4,
      170,
      23,
      30,
      3638,
      105,
      180,
    ],
    Array [
      10,
      "Audi",
      "A4 3.0 4dr",
      "Sedan",
      "Europe",
      "Front",
      31840,
      28846,
      3,
      6,
      220,
      20,
      28,
      3462,
      104,
      179,
    ],
    Array [
      11,
      "Audi",
      "A4 3.0 Quattro 4dr manual",
      "Sedan",
      "Europe",
      "All",
      33430,
      30366,
      3,
      6,
      220,
      17,
      26,
      3583,
      104,
      179,
    ],
    Array [
      12,
      "Audi",
      "A4 3.0 Quattro 4dr auto",
      "Sedan",
      "Europe",
      "All",
      34480,
      31388,
      3,
      6,
      220,
      18,
      25,
      3627,
      104,
      179,
    ],
    Array [
      13,
      "Audi",
      "A6 3.0 4dr",
      "Sedan",
      "Europe",
      "Front",
      36640,
      33129,
      3,
      6,
      220,
      20,
      27,
      3561,
      109,
      192,
    ],
    Array [
      14,
      "Audi",
      "A6 3.0 Quattro 4dr",
      "Sedan",
      "Europe",
      "All",
      39640,
      35992,
      3,
      6,
      220,
      18,
      25,
      3880,
      109,
      192,
    ],
    Array [
      15,
      "Audi",
      "A4 3.0 convertible 2dr",
      "Sedan",
      "Europe",
      "Front",
      42490,
      38325,
      3,
      6,
      220,
      20,
      27,
      3814,
      105,
      180,
    ],
    Array [
      16,
      "Audi",
      "A4 3.0 Quattro convertible 2dr",
      "Sedan",
      "Europe",
      "All",
      44240,
      40075,
      3,
      6,
      220,
      18,
      25,
      4013,
      105,
      180,
    ],
    Array [
      17,
      "Audi",
      "A6 2.7 Turbo Quattro 4dr",
      "Sedan",
      "Europe",
      "All",
      42840,
      38840,
      2.7,
      6,
      250,
      18,
      25,
      3836,
      109,
      192,
    ],
    Array [
      18,
      "Volvo",
      "S40 4dr",
      "Sedan",
      "Europe",
      "Front",
      25135,
      23701,
      1.9,
      4,
      170,
      22,
      29,
      2767,
      101,
      178,
    ],
    Array [
      19,
      "Audi",
      "A6 4.2 Quattro 4dr",
      "Sedan",
      "Europe",
      "All",
      49690,
      44936,
      4.2,
      8,
      300,
      17,
      24,
      4024,
      109,
      193,
    ],
    Array [
      20,
      "Audi",
      "A8 L Quattro 4dr",
      "Sedan",
      "Europe",
      "All",
      69190,
      64740,
      4.2,
      8,
      330,
      17,
      24,
      4399,
      121,
      204,
    ],
  ],
  "schema": Array [
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "_Index_",
      "type": "int",
      "width": 4,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Make",
      "type": "string",
      "width": 5,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Model",
      "type": "string",
      "width": 30,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Type",
      "type": "string",
      "width": 6,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Origin",
      "type": "string",
      "width": 6,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "DriveTrain",
      "type": "string",
      "width": 5,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "MSRP",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Invoice",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "EngineSize",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Cylinders",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Horsepower",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "MPG_City",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "MPG_Highway",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Weight",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Wheelbase",
      "type": "double",
      "width": 8,
    },
    Object {
      "attributes": Object {},
      "format": "",
      "label": "",
      "name": "Length",
      "type": "double",
      "width": 8,
    },
  ],
  "title": "Selected Rows from Table CARSJEST",
}
`;
